app/reducers/history.ts

```diff
@@ -0,0 +1,40 @@
+import { initialAppState } from './initialAppState';
+import { makeValidAddressBarUrl } from '$Utils/urlHelpers';
+import { TYPES as TABS_TYPES } from '$Actions/tabs_actions';
+import { logger } from '$Logger';
+
+const initialState = initialAppState.history;
+
+const updateHistory = ( state, payload ) => {
+    if ( payload.url ) {
+        const url = makeValidAddressBarUrl( payload.url );
+        if ( url !== state[state.length - 1] ) {
+            const newState = [...state];
+            newState.push( url );
+            return newState;
+        }
+        return state;
+    }
+    return state;
+};
+
+export function history( state: object = initialState, action ) {
+    const { payload } = action;
+
+    if ( action.error ) {
+        logger.error( 'ERROR IN ACTION', action.error );
+        return state;
+    }
+
+    switch ( action.type ) {
+        case TABS_TYPES.UPDATE_TAB: {
+            return updateHistory( state, payload );
+        }
+        case TABS_TYPES.TABS_RESET_STORE: {
```

Comment:
Should we use a tabs action here?

Should there be a general action for resetting the store if it can be applied across many reducers? (So maybe there's 'trigger_reset' which is the alias and 'reset_store' which can be passed into any reducer?